### PR TITLE
Allow trusted users to see whether a user is suspended

### LINKED ISSUE #19934: Allow trusted users to see whether a user is suspended
## Is this feature related to a problem? Please describe.

As a trusted user, I want to see whether a user has been suspended or not so that I know that admin action has already been enacted on their account.

## Describe the solution you’d like

- Trusted Users will be able to see whether a user has been suspended or not on their profile.

## Definition of Done

- [ ] For trusted users - show the "Suspended" pill user status (that is currently used in the member manager) on the suspended user's profile page.

## Additional Context

-The [user_profile_header area is a cached area](https://github.com/forem/forem/blob/Ridhwana/following-tag-page-buttons/app/views/users/show.html.erb#L20-L121) and we may not use current_user in a cached code path. Hence, we may want to make this an async call and insert it in the DOM afterwards.
- We are getting duplicate reports on user accounts from multiple trusted users who are unaware that action has already been taken. See this [slack thread](https://forem-team.slack.com/archives/CAA200SH5/p1691422492532989) (internal Forem Team only) for more context.
- Mockup of the "Suspended" pill 
image

### PR DESCRIPTION
**Pull request by jaw6**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

> As a trusted user, I want to see whether a user has been suspended or not so that I know that admin action has already been enacted on their account.

> We are getting duplicate reports on user accounts from multiple trusted users who are unaware that action has already been taken.

> The [user_profile_header area is a cached area](https://github.com/forem/forem/blob/Ridhwana/following-tag-page-buttons/app/views/users/show.html.erb#L20-L121) and we may not use current_user in a cached code path. Hence, we may want to make this an async call and insert it in the DOM afterwards.

As far as I know, we don't currently expose `suspended` status in an endpoint that's accessible in an API call from the front-end, so I needed to put that _somewhere_.

## Related Tickets & Documents

- Closes #19934 

## QA Instructions, Screenshots, Recordings

image

The "pill" looks a _little_ different here than on the admin panel, I have not _fully_ investigated (`css-is-awesome-mug.jpg`), but I think it might be close enough for the purpose here?

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] Are there any post deployment tasks we need to perform?

`users/show` is Rails-cached for `10.days`, but seems to also be cached in Fastly; to see the new feature appear on the site, we may need to manually bust this cache?

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by Ridhwana — APPROVED** [review]
Sorry I know I was not requested but I was in the neighbourhood reading the code so I thought I'd review 😅 This looks good to me 🚀

**Review by rt4914 — APPROVED** [review]

File: app/javascript/__tests__/asyncUserStatusCheck.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    beforeEach(() => {
+      fetch.mockResponse(fakeUserIsSuspended());
+    });
+
+    it('uses fetch to retrieve data async', async () => {
+      await asyncUserStatusCheck();
+      expect(fetch).toHaveBeenCalledWith('/users/123');
+    });
+
+    it('**does** badge the user', async () => {
```
Comment (dukegreene) [inline app/javascript/__tests__/asyncUserStatusCheck.test.js:69]: Love the emphasis here, as well as in your technical writing in general. I always get a sense of the tone of the speaker's voice through your written words.

File: app/javascript/packs/asyncUserStatusCheck.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import '@utilities/document_ready';
+
+export async function asyncUserStatusCheck() {
+  const indicator = ` <span data-testid="user-status" class="ml-3 c-indicator c-indicator--danger c-indicator--relaxed">Suspended</span>`;
+
+  const profile = document.querySelector('.profile-header__details');
+
+  if (profile && !profile.dataset.statusChecked) {
```
Comment (dukegreene) [inline app/javascript/packs/asyncUserStatusCheck.js:8]: Appreciating the care here to not throw too many requests around

File: app/javascript/utilities/document_ready.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Document.prototype.ready = new Promise((resolve) => {
```
[the remark this answers is not in the document]
Comment (dukegreene) [reply app/javascript/utilities/document_ready.js:1]: ++

**Review by dukegreene — APPROVED** [review]
LGTM! Clean conscientious changes across the stack

**Pull request merged**